### PR TITLE
ci: splitting release note into each CHANGELOG.md

### PR DESCRIPTION
**Pull request by suztomo**
The script splits the release note into multiple CHANGELOG.md in each module. This should run for Release Please pull request and add an additional commit to the pull request.

### CODE REVIEW

**Comment by suztomo** [comment]
Todo: resolve the date difference in golden file.

```
Run sh release-note-generation/golden_test.sh
Temporary directory created: /tmp/tmp.gjhKCf9oLU
--- /tmp/tmp.gjhKCf9oLU/testdata/java-analytics-admin/CHANGELOG.md	2023-06-13 03:01:12.520653793 +0000
+++ release-note-generation/golden/java-analytics-admin/CHANGELOG.md	2023-06-13 03:01:12.06065[4](https://github.com/googleapis/google-cloud-java/actions/runs/5250779717/jobs/9485073747?pr=9536#step:4:5)488 +0000
@@ -1,[6](https://github.com/googleapis/google-cloud-java/actions/runs/5250779717/jobs/9485073747?pr=9536#step:4:7) +1,6 @@
 # Changelog
 
-## 0.2[9](https://github.com/googleapis/google-cloud-java/actions/runs/5250779717/jobs/9485073747?pr=9536#step:4:10).0 (2023-06-13)
+## 0.29.0 (2023-06-12)
 
 ### Features
```

**Comment by suztomo** [comment]
Next step: Integrate the script to https://github.com/googleapis/google-cloud-java/blob/main/.github/workflows/changelog_generation.yaml#L6.

File: .github/release-note-generation/golden/java-analytics-admin/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+* Add the resource definition of a STT recognizer ([#9492](https://github.com/googleapis/google-cloud-java/issues/9492)) ([f2baa92](https://github.com/googleapis/google-cloud-java/commit/f2baa928c4f4d9959591ea905d7c9042320dbc71))
+
+### Dependencies
+
+* **deps:** update dependency com.google.cloud:google-cloud-shared-config to v1.5.6 ([#9494](https://github.com/googleapis/google-cloud-java/issues/9494)) ([739763f](https://github.com/googleapis/google-cloud-java/commit/739763f15ffa2434d7fa089899165f1e8fa5f870))
+* **deps:** update dependency com.google.cloud:google-cloud-shared-dependencies to v3.11.0 ([#9505](https://github.com/googleapis/google-cloud-java/issues/9505)) ([30bc6f0](https://github.com/googleapis/google-cloud-java/commit/30bc6f0aef5b95549230dbd5b5246f2a8dab4ba4))
+
+
+## [0.15.0](https://github.com/googleapis/google-cloud-java/compare/google-analytics-admin-v0.14.1-SNAPSHOT...google-analytics-admin-v0.15.0) (2022-10-24)
```
Comment (burkedavison) [inline .github/release-note-generation/golden/java-analytics-admin/CHANGELOG.md:15]: Very nice PR. These legacy diff links aren't working, though perhaps out of scope for this PR.

Comment (suztomo) [reply .github/release-note-generation/golden/java-analytics-admin/CHANGELOG.md:15]: Right.The diff links won't work for this monorepo. We can compare different tags but the files involved include irrelevant files outside a specific module.

**Review by ddixit14 — APPROVED** [review]

**Pull request merged**